djunicode/ergo
# Added the initialisation function

## Description

> Initialisation Function is set up to get various information about the host system, such as if various programs are installed : node, npm python, etc , check for which platform it is running.

> Testing is done using ```Chai``` and ```Mocha```

## Changes

- Added file ```initial.js``` at location ```app/electron/```
- Added ```test``` folder which contains tests in ```chai``` and ```mocha```
- Added ```initialisationFunction.js``` file for testing at location ```test/```
- Dependencies Installed  : ```chai``` , ```chai-http``` , ```mocha``` , ```shelljs``` , ```electron-store```

## Issue

> Closes issue :  #4

## Impacted Area

- The ```config.json``` file containing information after testing is stored at : 
  ```C:\Users\[USER_NAME]\AppData\Roaming\electron-store-nodejs```

- The ```config.json``` file containing information during the initialisation of the electron app is stored at : 
  ```C:\Users\[USER_NAME]\AppData\Roaming\ergo```

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

- require the function from ```app/electron/initial.js``` into ```app/electron/main.js``` and then call the function inside ```main.js``` , run the app ```npm run dev```
- Check the ```config.json``` file stored at :  ```C:\Users\[USER_NAME]\AppData\Roaming\ergo``` 
- To run ```test``` : ```npm run test test/initialisationFunction```
- Check the ```config.json``` file stored at : ```C:\Users\[USER_NAME]\AppData\Roaming\electron-store-nodejs```

## Screenshots
- Screenshot of Testing : 

> ![image](https://user-images.githubusercontent.com/68227858/106849610-00a9da80-66d9-11eb-9cfb-97ef02f053db.png)


- Screenshot of the config.json file : 

> ![image](https://user-images.githubusercontent.com/68227858/106849741-4797d000-66d9-11eb-9467-e5d77f3fe844.png)
